examples/generative/corrdiff/datasets/cwb.py

```diff
@@ -472,34 +447,34 @@ def __len__(self):
     def longitude(self):
         """Get longitude values from the dataset."""
         lon = self._dataset.longitude()
-        return lon if self.train else lon[..., : self.crop_size_y, : self.crop_size_x]
+        return lon if self.train else lon[..., : self.img_shape_y, : self.img_shape_x]
 
     def latitude(self):
         """Get latitude values from the dataset."""
         lat = self._dataset.latitude()
-        return lat if self.train else lat[..., : self.crop_size_y, : self.crop_size_x]
+        return lat if self.train else lat[..., : self.img_shape_y, : self.img_shape_x]
```

Comment:
**For later**

Not new code, but it is odd for a few reasons
1. different answers based on training vs not. I understand why, but to me lat is lat.
2. the crops always start at (0,0) is the also true in the __getitem__ below?

Overall, for the case where lat/lon may depend on patching/cropping etc, I think it's easier if these coordinate values are returned alongside the data or if you modify these coordinates values to take global_index as an argument.

```
def latitude(self, global_index):
    ...
```

Reply:
In the previous version, starting location is defined by a random number generator if crop size is smaller than image shape, which seems to be a prototype of patch-based training. In the future we could consider move the patching operation back to dataloader and let the loader return coordinate values. The patching operation for regression's output could be performed based on returned global indices. I have added this to issue #448 as a reminder. Resolving this conversation for now.